File: Reqnroll/Chapter13/13.3-IntroduceAndReferPattern/After/WIMP.Specs/StepDefinitions/AuthenticationStepDefinitions.cs
using Reqnroll;

using WIMP.App.Services;
using WIMP.Specs.Support;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class AuthenticationStepDefinitions(AuthenticationContext authContext)
{
    [Given("the customer {string} is authenticated")]
    public void GivenTheCustomerIsAuthenticated(string customerName)
    {
        AuthenticationService.Login(customerName);
        authContext.AuthenticatedCustomerName = customerName;
    }
}

File: CucumberJVM/Chapter13/13.2-ShareableScenarioContextPattern/Before/src/test/java/com/wimp/app/specs/stepdefinitions/AuthenticationStepDefinitions.java
package com.wimp.app.specs.stepdefinitions;

import com.wimp.app.services.AuthenticationService;
import io.cucumber.java.en.*;

public class AuthenticationStepDefinitions {
    @Given("the customer {string} is authenticated")
    public void theCustomerIsAuthenticated(String customerName) {
        AuthenticationService.login(customerName);
    }
}
